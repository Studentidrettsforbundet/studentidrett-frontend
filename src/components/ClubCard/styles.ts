import { style, classes } from 'typestyle';
import colors from '../../styles/colors';


export const clubCard = style({
        padding: 0,
    
});

export const cardHeader = style({
    color: colors.cardHeader,
    fontSize: '19px',
    fontFamily: 'ArchivoSemiBold',
    textAlign: 'left',
    marginBottom: '10px'

});

export const cardBody = style({
    backgroundColor: 'inherit',
    color: colors.cardBodyText,
    padding: '15px',
    fontSize: '16px',
    textAlign: 'left',

});