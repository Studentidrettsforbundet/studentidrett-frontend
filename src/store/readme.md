# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
