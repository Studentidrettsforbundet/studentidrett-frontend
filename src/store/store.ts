import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
