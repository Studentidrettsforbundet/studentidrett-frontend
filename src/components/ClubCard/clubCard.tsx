import React from 'react';
import { useDispatch } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';
import { clubInterface } from '../../interfaces';
import { resetFetchStatusesActionCreator } from '../../store/thunks/thunkActions';
import image from '../../assets/placeholder.png';
import { card, cardImg } from '../../styles/card';
import { clubCard, cardHeader, cardBody} from './styles';
import {classes} from 'typestyle';

const ClubCard = (club: clubInterface) => {
    const dispatch = useDispatch();
    return (
        <Link
            to={`/clubs/${club.id}`}
            key={club.id}
            className={'unstyled_link'}
            onClick={() => dispatch(resetFetchStatusesActionCreator())}
        >
            <div className={classes(card, clubCard)}>
                    <img className={cardImg} src={image} alt="Club" />
                <div className={cardBody}>
                    <div className={cardHeader}>
                        {club.name}
                    </div>
                    <p>{club.description.slice(0,115)} ...</p>
                </div>
            </div>
        </Link>
    );
};

export default ClubCard;
