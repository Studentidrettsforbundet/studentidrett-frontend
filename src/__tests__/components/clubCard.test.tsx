import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import ClubCard from '../../components/ClubCard/clubCard';
import store from '../../store/store';

describe('clubCard', () => {
    test('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/MidtNorge/Fotball']}>
                    <ClubCard
                        id={0}
                        city={0}
                        name={'NTNUIFotball'}
                        description={'This is where the description goes'}
                        contact_email={'contact.email.com'}
                        membership_fee={'100'}
                        register_info={'this is where register info goes'}
                    />
                </MemoryRouter>
            </Provider>,
        );
        expect(tree).toMatchSnapshot();
    });
});
