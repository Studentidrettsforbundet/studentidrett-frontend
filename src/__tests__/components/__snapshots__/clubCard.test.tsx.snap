// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clubCard renders correctly 1`] = `
<a
  className="unstyled_link"
  href="/clubs/3"
  onClick={[Function]}
>
  <div
    className="f1i8l76j"
  >
    <div
      className="f2lk941"
    >
      <img
        alt="Club"
        className="fb12b09"
        src="placeholder.png"
      />
    </div>
    <div
      className="f1drtrr0"
    >
      <h5
        className="fm5b097"
      >
        NTNUI Bowling
      </h5>
      <p>
        
      </p>
    </div>
  </div>
</a>
`;
